packages/teleterm/src/ui/services/clusters/clustersService.ts

```diff
@@ -383,6 +393,95 @@ export class ClustersService extends ImmutableStore<ClustersServiceState> {
     }
   }
 
+  async fetchServers(params: ServerSideParams) {
```

Comment:
The callsites that call these new methods from `clustersService` need to wrap the calls to those methods in `retryWithRelogin`.

When I came back to the running dev version of Electron, I saw this:

![cert has expired](https://user-images.githubusercontent.com/27113/193069136-57157289-bfde-46ed-bda7-b8d6f33e2f57.png)

I think this couldn't happen by itself because AFAIK Connect just doesn't decide to make a request by itself, I must've went from the view request "page" to the list of all requests.

In any case, the general rule is that the backend handlers that use `RetryWithRelogin` need to be wrapped with `retryWithRelogin` on the frontend for the whole system to take effect.

Look up how other places in the frontend app do this. `retryWithRelogin` needs to tightly wrap the call to ClustersService.

You might have a problem with passing a meaningful value as the `resourceUri` argument to `retryWithRelogin`. In case of, for example, listing all resources, I don't think you'll have a meaningful resource URI to pass there. In that case, that argument is used only to get the root cluster URI, so you can just pass the cluster URI (if it's a leaf cluster URI, `retryWithRelogin` will extract the root cluster URI from it automatically).

Reply:
Ok, i wrapped the requests. Funnily enough, that `fetchServers` seems to have been vestigial from before the `getAllServers' and 'getServers' distinction as it isn't used anywhere in the codebase atm.